Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingAccelerationStructurePass.cpp

```diff
@@ -202,6 +275,85 @@ namespace AZ
             context.GetCommandList()->BuildTopLevelAccelerationStructure(*rayTracingFeatureProcessor->GetTlas());
 
             ++m_frameCount;
+
+            EndScopeQuery(context);
+        }
+
+        void RayTracingAccelerationStructurePass::AddScopeQueryToFrameGraph(AZ::RHI::FrameGraphInterface frameGraph)
+        {
+            const auto addToFrameGraph = [&frameGraph](AZ::RHI::Ptr<AZ::RPI::Query> query)
+            {
+              query->AddToFrameGraph(frameGraph);
+            };
+
+            ExecuteOnTimestampQuery(addToFrameGraph);
+            ExecuteOnPipelineStatisticsQuery(addToFrameGraph);
+        }
+
+        void RayTracingAccelerationStructurePass::BeginScopeQuery(const AZ::RHI::FrameGraphExecuteContext& context)
+        {
+            const auto beginQuery = [&context, this](AZ::RHI::Ptr<AZ::RPI::Query> query)
+            {
+              if (query->BeginQuery(context) == AZ::RPI::QueryResultCode::Fail)
+              {
+                  AZ_UNUSED(this); // Prevent unused warning in release builds
+                  AZ_WarningOnce(
+                      "RayTracingAccelerationStructurePass", false,
+                      "BeginScopeQuery failed. Make sure AddScopeQueryToFrameGraph was called in SetupFrameGraphDependencies"
+                      " for this pass: %s",
+                      this->RTTI_GetTypeName());
+              }
+            };
+
+            if (context.GetCommandListIndex() == 0)
+            {
+                ExecuteOnTimestampQuery(beginQuery);
+                ExecuteOnPipelineStatisticsQuery(beginQuery);
+            }
+        }
+
+        void RayTracingAccelerationStructurePass::EndScopeQuery(const AZ::RHI::FrameGraphExecuteContext& context)
+        {
+            const auto endQuery = [&context](AZ::RHI::Ptr<AZ::RPI::Query> query)
+            {
+              query->EndQuery(context);
+            };
+
+            // This scope query implementation should be replaced by
+            // [ATOM-5407] [RHI][Core] - Add GPU timestamp and pipeline statistic support for scopes
```

Comment:
Dont add new comments with JIRA numbers (i.e [ATOM-5407]) as no one outside Amazon will get the context. Feel free to add a GHI and tag that here if you want.

Reply:
I have added a new issue https://github.com/o3de/o3de/issues/16945 and amended the comment to include it inside the code.